R/final_size.R

```diff
@@ -1,54 +1,77 @@
 #' Calculate final epidemic size
 #'
-#' This function calculates final epidemic size using SIR model for heterogeneously mixing population
+#' This function calculates final epidemic size using SIR model for
+#' heterogeneously mixing population
 #' @param r0  Basic reproduction number. Default is 2
-#' @param contact_matrix  Social contact matrix. Entry mm_ij gives average number of contacts in group i reported by participants in group j
-#' @param demography_vector  Demography vector. Entry pp_i gives proportion of total population in group i (model will normalise if needed)
-#' @param prop_suscep  Proportion of each group susceptible. Null assumption is fully susceptible
+#' @param contact_matrix  Social contact matrix. Entry $mm_{ij}$ gives average
+#' number of contacts in group $i$ reported by participants in group j
+#' @param demography_vector  Demography vector. Entry $pp_{i}$ gives proportion
+#' of total population in group $i$ (model will normalise if needed)
+#' @param prop_suscep  Proportion of each group susceptible. Null assumption is
+#' fully susceptible
 #' @keywords epidemic model
 #' @export
 #' @examples
-#' final_size()
+#' library(socialmixr)
+#' data(polymod)
+#' contact_data <- contact_matrix(polymod, countries = "United Kingdom", age.limits = c(0,20,40))
+#' 
+#' c_matrix <-t(contact_data$matrix) # Define contact matrix (entry ij is contacts in group i reported by group j)
+#' d_vector <- contact_data$participants$proportion # Define proportion in each age group
+#' p_suscep <- c(1,0.5,0.5) # Define proportion of age group that is susceptible to infection
+#' 
+#' # Run final size model
+#' final_size(r0=2,contact_matrix = c_matrix, demography_vector = d_vector, prop_suscep = p_suscep)
+#'
+final_size <- function(r0 = 2, contact_matrix, demography_vector,
+                       prop_suscep = NULL) {
 
-final_size<-function(r0=2,contact_matrix,demography_vector,prop_suscep=NULL){
-  
   # Check inputs
-  if(is.null(prop_suscep)){prop_suscep <- demography_vector*0 + 1} # Assume fully susceptible if no entry
-  if(length(contact_matrix[1,])!=length(demography_vector)){stop("demography vector needs to be same size as contact matrix")}
-  if(length(demography_vector)!=length(prop_suscep)){stop("demography vector needs to be same size as susceptibility vector")}
-  pp0=as.numeric(demography_vector/sum(demography_vector))
-  
+  if (is.null(prop_suscep)) {
+    prop_suscep <- demography_vector * 0 + 1
+  } # Assume fully susceptible if no entry
+  if (length(contact_matrix[1, ]) != length(demography_vector)) {
+    stop("demography vector needs to be same size as contact matrix")
+  }
+  if (length(demography_vector) != length(prop_suscep)) {
+    stop("demography vector needs to be same size as susceptibility vector")
+  }
```

Comment:
I find it easier to check multiple assumptions about inputs using `checkmate`. Here for instance, there seems to be an assumption that `contact_matrix` is a matrix, but this is not formally checked.

Reply:
Added as #23